### PR TITLE
MEN-3744: add streaming, resume and compression to overview/artifact

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

File: 02.Overview/03.Artifact/docs.md [hunk 1]
```diff
@@ -79,6 +79,26 @@ Not all Artifacts have dependencies; if they don't, then it doesn't matter what
...
+
+The tar format supports streaming, which Mender takes advantage of. As a Mender
+Artifact is downloaded from the Mender server or external storage, the Mender
+client streams the root file system within it directly to the inactive partition,
+without needing any temporary storage for unpacking it before it is written.
+This drastically reduces storage requirements for the update process,
+improves performance and reduces flash wear.
+
+In cases where Artifact downloads are interrupted, e.g. due to unreliable wireless
+network connectivity, Mender will resume the download from where it was
```
Comment (oleorhagen) [inline 02.Overview/03.Artifact/docs.md:92]: Suggestion:
`s/Mender/the Mender client/`

File: 02.Overview/03.Artifact/docs.md [hunk 2]
```diff
@@ -79,6 +79,26 @@ Not all Artifacts have dependencies; if they don't, then it doesn't matter what
...
+interrupted, using [HTTP range requests](https://tools.ietf.org/html/rfc7233?target=_blank).
+
+To enable streaming and control based on metadata, like aborting the download
```
Comment (oleorhagen) [inline 02.Overview/03.Artifact/docs.md:95]: Nit:
It's not just the payload which is compressed, some of the metadata are too, like the headers

**Review by oleorhagen — APPROVED** [review]
Great! Thanks :smile:

File: 02.Overview/03.Artifact/docs.md (hunk 2 above)
Comment (kacf) [reply 02.Overview/03.Artifact/docs.md:95]: Yes, everything is compressed except the outer tar meta-data, and the version header. All of the Artifact metadata is compressed.

**Review by estenberg — APPROVED** [review]
Thanks for knocking this off.

It's fine to keep it here, not worth spending more time on I think.

**Pull request merged**